### PR TITLE
Add support for required easyjson annotation

### PR DESCRIPTION
**Pull request by Raynos**
It's useful for the go structs generated by thriftrw to support
the `required` go tag annotation that easyjson uses to generate
"validating" json parser.

The third party library [easyjson](https://github.com/mailru/easyjson)
supports the notion of [`required tag`](https://github.com/mailru/easyjson/blob/master/tests/data.go#L667-L670)

This allows a user of thriftrw that uses JSON encoding ( instead of thrift
binary encoding ) to be able to validate that all required fields exist
when doing the parsing stage.

r: @abg

### CODE REVIEW

File: gen/field.go [hunk 1]
```diff
@@ -161,6 +161,10 @@ func compileJSONTag(f *compile.FieldSpec, name string, opts ...string) *structta
 		t.Options = append(t.Options, "omitempty")
 	}
 
+	if f.Required {
+		t.Options = append(t.Options, "required")
```
Comment (abhinav) [inline gen/field.go]: Will this continue to work if I was doing, `(go.tag = "json:\"required\"")`? Can you add a test for this?

(You may want to check `HasOption` like the previous block.)

Comment (Raynos) [inline gen/field.go]: yeah will test / verify.

Comment (Raynos) [inline gen/field.go]:
```
 type Rename struct {
-       Default   string `json:"default,required"`
+       Default   string `json:"default,required,required"`
        CamelCase string `json:"snake_case,required"`
 }
```

lol.

**Comment by Raynos** [comment]
Updated with more tests.

**Comment by abhinav** [comment]
FYI @alextbok in case you wanted a look.

**Pull request merged**